README.md

```diff
@@ -64,14 +74,173 @@ If you have a suggestion for a feature, please feel free to file it under "issue
+
+    https://appsforoffice.microsoft.com/lib/1/hosted/office.js
+
+If the library is available as an NPM package, you can just reference it by name:
+
+    spectrum-colorpicker/spectrum.js
+    spectrum-colorpicker/spectrum.css
+
+This will use <https://unpkg.com/>, a global CDN for NPM libraries, to substitute in the appropriate files. Thus, the above is equivalent to:
+
+    https://unpkg.com/spectrum-colorpicker@1.8.0/spectrum.js
+    https://unpkg.com/spectrum-colorpicker@1.8.0/spectrum.css
+
+> Tip: To see the available files in a package, simply include a `/` after the package name, and you'll get a directory listing.  E.g., <https://unpkg.com/spectrum-colorpicker/>. It will also conveniently tell you the available version numbers, via a dropdown at the top right of the site:
+>
+> ![Import](.github/images/unpkg.jpg)
+
+
+#### Moderate: NPM package is an AMD module, but the library is also available via a "dist" build
+
+Suppose you add a reference to the library (say, the [javascript-qrcode](https://www.npmjs.com/package/javascript-qrcode) NPM package), but all you get is an error in the console:
+
+    ["Script error",
+     "https://unpkg.com/javascript-qrcode",
+     0, 0, undefined]
+
+In Internet Explorer (and hence on Add-ins on a PC version of Office), you don't get much more information than this -- but on a different platform (or Office Online on a different browser), you will see:
+
+    ["Uncaught ReferenceError: module is not defined",
+     "https://unpkg.com/javascript-qrcode@1.0.7",
+     3302, 1, ReferenceError: module is not defined {}]
+
+If you are lucky, even though the *NPM* package doesn't contain a web-ready version of the package, the web-compatible version may be available elsewhere.  In the case of this `javascript-qrcode` package, the [NPM entry](https://www.npmjs.com/package/javascript-qrcode) points to a GitHub repo, <https://github.com/siciarek/javascript-qrcode>. And the repo, in turn, has a "dist" folder (convention), along with exampe usage that also shows `<script src="dist/qrcode.min.js"></script>`.
+
+This is great news, as it means that a web-ready version of the library is available!  Just click on the "dist" folder, and find the file you want:  <https://github.com/siciarek/javascript-qrcode/blob/master/dist/qrcode.min.js>
+
+Almost there!  To be consumed inside of Script Lab, we need the file to be served in *RAW* form, ***and*** with correct Content-Type headers.  To do this, use a service like [RawGit](https://rawgit.com) to transform the file for you, and give you a URL you can use directly.
+
+In this case, pasting in the above URL into RawGit produces:
+
+    https://cdn.rawgit.com/siciarek/javascript-qrcode/041ad555/dist/qrcode.min.js
+
+Include this in your Libraries tab, and you're good to go!
+
+
+#### Harder: NPM package is an AMD module (must be loaded with `require.js`), and is *not* available in "dist" form
```

Comment:
Technically this is the `medium` difficulty one. Given that there's a possibility of loading stuff in.

@Zlatkovsky a todo would be adding a require.config tab when we see require.js being a dependency.